### PR TITLE
feat: support TS 5.1, drop TS 4.7

### PR DESCRIPTION
**Pull request by ephys**
## Description Of Change

Dropped support for TS 4.6

Added support for TS 5.1

## Todos

- will need to update which version of typescript is accepted in our documentation

### CODE REVIEW

File: packages/core/src/hooks-legacy.ts
Comment (WikiRik) [inline packages/core/src/hooks-legacy.ts:66]: Shouldn't this be 5.0?

Comment (ephys) [reply packages/core/src/hooks-legacy.ts]: yep! fixed

**Comment by ephys** [comment]
Looks like this will be blocked until typedoc updates

**Review by WikiRik — COMMENTED** [review]
Looks good, PR has been raised in typedoc so once that has made a new release we can merge this as well

**Review by WikiRik — APPROVED** [review]

**Pull request merged**
